primitives/session/src/lib.rs

```diff
@@ -44,6 +47,33 @@ sp_api::decl_runtime_apis! {
 		/// Returns the list of public raw public keys + key type.
 		fn decode_session_keys(encoded: Vec<u8>) -> Option<Vec<(Vec<u8>, KeyTypeId)>>;
 	}
+
+	/// Historical session membership runtime api.
+	pub trait SessionMembership {
```

Comment:
Maybe we can just hide this in a GRANDPA runtime API to avoid leaking runtime-internal details about the proof. After all, I don't think node-side code will inspect this proof deeply.

Reply:
Done.